saleor/payment/utils.py

```diff
@@ -107,8 +248,89 @@ def create_payment_information(
         checkout_metadata=checkout_metadata,
         payment_metadata=payment.metadata,
         psp_reference=payment.psp_reference,
+        refund_data=refund_data,
+        _resolve_lines=lambda: create_payment_lines_information(
+            payment, manager or get_plugins_manager()
+        ),
+    )
+
+
+RefundLines = Iterator[Tuple[Any, OrderLine]]
+
+
+def _prepare_refund_lines(
+    order: Order,
+    order_lines_to_refund: List[OrderLineInfo],
+    fulfillment_lines_to_refund: List[FulfillmentLineData],
+) -> Iterator[Tuple[int, int]]:
+    previous_fulfillment_lines = FulfillmentLine.objects.prefetch_related(
+        "order_line"
+    ).filter(
+        fulfillment__order_id=order.pk,
+        fulfillment__status__in=[
+            FulfillmentStatus.REFUNDED,
+            FulfillmentStatus.REFUNDED_AND_RETURNED,
+        ],
+        order_line__variant_id__isnull=False,
     )
 
+    previous_refund_lines = (
+        (p_variant_id, line.quantity)
+        for line in previous_fulfillment_lines
+        if (p_variant_id := line.order_line.variant_id)
+    )
+
+    current_order_refund_lines = (
+        (variant.id, line.quantity)
+        for line in order_lines_to_refund
+        if (variant := line.variant)
+    )
+
+    current_fulfillment_refund_lines = (
+        (f_variant_id, line.quantity)
+        for line in fulfillment_lines_to_refund
+        if (f_variant_id := line.line.order_line.variant_id)
+    )
+
+    return chain(
+        previous_refund_lines,
+        current_order_refund_lines,
+        current_fulfillment_refund_lines,
+    )
+
+
+def create_refund_data(
+    order: Order,
+    order_lines_to_refund: List[OrderLineInfo],
+    fulfillment_lines_to_refund: List[FulfillmentLineData],
+    refund_shipping_costs: bool,
+) -> Dict[int, int]:
+    order_lines = {line.variant_id: line.quantity for line in order.lines.all()}
+
+    refund_lines = _prepare_refund_lines(
+        order, order_lines_to_refund, fulfillment_lines_to_refund
+    )
+
+    summed_refund_lines: Dict[int, int] = defaultdict(int)
```

Comment:
Looks like, you're calling `_prepare_refund_lines` function only to get a data to iterate over them. 
What about moving this for to `_prepare_refund_lines`? Instead of creating `previous_refund_lines`, `current_order_refund_lines` and `current_fulfillment_refund_lines`, you could immedietly create `refund_lines: Dict[int, int] = defaultdict(int)` and fulfill it with all data passed to `_prepare_refund_lines`.

Reply:
Resolved in recent commit.